driver-mapping/src/test/java/com/datastax/driver/mapping/MappingConfigurationNamingStrategyTest.java

```diff
@@ -0,0 +1,153 @@
+import com.datastax.driver.mapping.annotations.PartitionKey;
+import com.datastax.driver.mapping.annotations.Table;
+import org.testng.annotations.Test;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class MappingConfigurationNamingStrategyTest extends CCMTestsSupport {
+
+    private static AtomicInteger counter = new AtomicInteger(0);
+
+    @Override
+    public void onTestContextInitialized() {
+        execute("CREATE TABLE lowerlisp (\"primary-key\" int primary key, \"my-value\" int)");
+        execute("CREATE TABLE uppersnake (\"PRIMARY_KEY\" int primary key, \"MY_VALUE\" int)");
+    }
+
+    @Table(name = "lowerlisp")
+    static class UpperSnake {
+
+        @Column(caseSensitive = true)
+        @PartitionKey
+        int PRIMARY_KEY;
+
+        @Column(caseSensitive = true)
+        int MY_VALUE;
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+
+            UpperSnake that = (UpperSnake) o;
+
+            return PRIMARY_KEY == that.PRIMARY_KEY && MY_VALUE == that.MY_VALUE;
+        }
+
+        @Override
+        public int hashCode() {
+            int result = PRIMARY_KEY;
+            result = 31 * result + MY_VALUE;
+            return result;
+        }
+    }
+
+    @Test(groups = "short")
+    public void should_use_naming_strategy_fields() {
+        MappingConfiguration conf = MappingConfiguration.builder()
+                .withPropertyMapper(new DefaultPropertyMapper()
+                        .setNamingStrategy(new DefaultNamingStrategy(new CommonNamingConventions.UpperSnakeCase(),
+                                new CommonNamingConventions.LowerLispCase()))
+                        .setPropertyAccessStrategy(PropertyAccessStrategy.FIELDS))
+                .build();
+
+        MappingManager mappingManager = new MappingManager(session(), conf);
+        Mapper<UpperSnake> mapper = mappingManager.mapper(UpperSnake.class);
+
+        UpperSnake in = new UpperSnake();
+        in.PRIMARY_KEY = counter.incrementAndGet();
+        in.MY_VALUE = counter.incrementAndGet();
+
+        mapper.save(in);
+
+        UpperSnake out = mapper.get(in.PRIMARY_KEY);
+        assertThat(out).isEqualTo(in);
+    }
+
+    @Table(name = "uppersnake")
+    @SuppressWarnings({"unused", "WeakerAccess"})
+    static class LowerSnake {
+
+        private int _primaryKey;
+        private int _myValue;
+
+        @Column(caseSensitive = true)
+        @PartitionKey
+        public int getprimary_key() {
+            return _primaryKey;
+        }
+
+        public void setprimary_key(int k) {
+            this._primaryKey = k;
+        }
+
+        @Column(caseSensitive = true)
+        public int getmy_value() {
+            return _myValue;
+        }
+
+        public void setmy_value(int v) {
```

Comment:
Hrmm maybe this is taking things a bit too far.  I guess if a user wants to do that they could make their own naming convention.   Also I expect users to more likely use field based entities instead of getter/setters when using custom naming strategies.